plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -365,10 +365,8 @@ else if (type instanceof ScaledFloatType) {
             return new TypeAndDecoder(DOUBLE, new DoubleDecoder.Descriptor(path));
         }
         else if (type instanceof DateTimeType) {
-            if (((DateTimeType) type).getFormats().isEmpty()) {
-                return new TypeAndDecoder(TIMESTAMP_MILLIS, new TimestampDecoder.Descriptor(path));
-            }
-            // otherwise, skip -- we don't support custom formats, yet
+            List<String> formats = ((DateTimeType) type).getFormats();
```

Comment:
Use enhanced pattern variable.

Reply:
Execuse me, what does "enhanced pattern variable" mean?

Reply:
`else if (type instanceof DateTimeType dateTimeType) {`

It's already fixed in master though. Please rebase.